refactor(compiler): move test/test_bindings to testing/test_bindings

jasonchoimtt opened the pull request:
`test_bindings` is used in core test cases too, but `test` should be private to the package, so it should live in `testing`.

//cc: @IgorMinar 

**Please check if the PR fulfills these requirements**
- The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- Tests for the changes have been added (for bug fixes / features)
- Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
Refactoring (no functional changes, no api changes)
```

**Does this PR introduce a breaking change?** (check one with "x")

```
No
```

IgorMinar commented on modules/@angular/compiler/testing/test_bindings.ts:
```diff
@@ -6,10 +6,12 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
```
can you add a todo comment that this is here just so that we can access it from various internal test packages that we should really either get rid of it or create a separate package "@angular/internal_testing" and move stuff like this there. thanks

jasonchoimtt commented:
Does the comment look right now?

Commit:
refactor(compiler): move test/test_bindings to testing/test_bindings

`test_bindings` is used in core test cases too, but `test` should be
private to the package, so it should live in `testing`.

jasonchoimtt commented:
@IgorMinar please check

kara commented:
@jasonchoimtt  Can you rebase?

jasonchoimtt commented:
done!
